kivy/ext/mpl/backend_kivy.py

```diff
@@ -0,0 +1,1013 @@
+from matplotlib.mathtext import MathTextParser
+from matplotlib import rcParams
+from hashlib import md5
+from matplotlib import _png
+from matplotlib.font_manager import weight_as_number
+from matplotlib import _path
+
+try:
+    import kivy
+except ImportError:
+    raise ImportError("this backend requires Kivy to be installed.")
+
+from kivy.app import App
+from kivy.graphics.texture import Texture
+from kivy.graphics import Rectangle
+from kivy.uix.widget import Widget
+from kivy.uix.label import Label
+from kivy.uix.floatlayout import FloatLayout
+from kivy.uix.behaviors import FocusBehavior
+from kivy.uix.actionbar import ActionBar, ActionView, \
+                                ActionButton, ActionToggleButton, \
+                                ActionPrevious, ActionOverflow, ActionSeparator
+from kivy.base import EventLoop
+from kivy.core.text import Label as CoreLabel
+from kivy.graphics import Color, Line
+from kivy.graphics import Rotate, Translate
+from kivy.graphics.instructions import InstructionGroup
+from kivy.graphics.tesselator import Tesselator
+from kivy.graphics.context_instructions import PopMatrix, PushMatrix
+from kivy.logger import Logger
+from kivy.graphics import Mesh
+from kivy.resources import resource_find
+from kivy.uix.stencilview import StencilView
+from kivy.core.window import Window
+from kivy.uix.button import Button
+from kivy.uix.boxlayout import BoxLayout
+from kivy.uix.floatlayout import FloatLayout
+from kivy.uix.relativelayout import RelativeLayout
+from kivy.uix.popup import Popup
+from kivy.properties import ObjectProperty
+from kivy.clock import Clock
+from kivy.uix.textinput import TextInput
+from kivy.lang import Builder
+from kivy.logger import Logger
+
+import numpy as np
+import io
+import textwrap
+from functools import partial
+from math import cos, sin, pi
+
+try:
+    kivy.require('1.9.0')  # I would need to check which release of
+    # Kivy would be the best suitable.
+except AttributeError:
+    raise ImportError(
+        "kivy version too old -- it must have require_version")
+
+
+class SaveDialog(FloatLayout):
+    save = ObjectProperty(None)
+    text_input = ObjectProperty(None)
+    cancel = ObjectProperty(None)
+
+
+class MPLKivyApp(App):
+    '''Creates the App initializing a FloatLayout with a figure and toolbar
+       widget.
+    '''
+    figure = ObjectProperty(None)
+    toolbar = ObjectProperty(None)
+
+    def build(self):
+        EventLoop.ensure_window()
+        layout = FloatLayout()
+        if self.figure:
+            self.figure.size_hint_y = 0.9
+            layout.add_widget(self.figure)
+        if self.toolbar:
+            self.toolbar.size_hint_y = 0.1
+            layout.add_widget(self.toolbar)
+        return layout
+
+
+def _create_App(fig_canvas, toolbar):
+    '''Method to instantiate a MPLKivyApp.
+    '''
+    app = App.get_running_app()
+    if app is None:
+        app = MPLKivyApp(figure=fig_canvas, toolbar=toolbar)
```

Comment:
This doesn't make sense here. You're creating a app here, but not actually doing anything with it. Then you run the current app, in show's mainloop. At that point, the app that is run could be any app, even one created by the user later. Instead, you need to create and run the app in mainloop.